src/ucs/sys/iovec.inl

```diff
@@ -79,17 +78,16 @@
             } \
             \
             ucs_assert(__remain_length >= __dst_iov_length); \
-            __total_length             += __dst_iov_length; \
             __remain_length            -= __dst_iov_length; \
             ++__dst_iov_it; \
             \
         } \
         \
-        ucs_assert((__total_length <= _max_length) && \
+        ucs_assert((__remain_length >= 0) && \
```

Comment:
this will always be true, because `__remain_length` is `size_t`

Reply:
good catch!
fixed